### PR TITLE
feat(source): add more types for avro

### PR DESCRIPTION
**Pull request by waruto210**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- add `timestamp-micros` and `duration` support for avro

## Checklist

~- [] I have written necessary rustdoc comments~
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Connector (sources & sinks)

### Release note

user can use `timestamp-micros` and `duration` type in avro schema:
```json
{
  "name": "anniversary",
  "type": "long",
  "logicalType": "timestamp-micros",
  "default": null
},
{
  "name": "passed",
  "type": {
    "name": "interval",
    "type": "fixed",
    "size": 12
  },
  "logicalType": "duration"
}
```

## Refer to a related PR or issue link (optional)

None

### CODE REVIEW

File: e2e_test/source/kafka.slt [hunk 1]
```diff
@@ -293,9 +293,9 @@ select id, first_name, last_name, email from s8;
 1005 add add2 add
 
 query IITFFBTT
-select id, sequence_id, name, score, avg_score, is_lasted, entrance_date, birthday from s9;
+select id, sequence_id, name, score, avg_score, is_lasted, entrance_date, birthday, passed from s9;
 ----
-32 64 str_value 32 64 t 1970-01-01 1970-01-01 00:00:00
+32 64 str_value 32 64 t 1970-01-01 1970-01-01 00:00:00 1 mon 1 day 00:00:01
```
Comment (tabVersion) [inline e2e_test/source/kafka.slt:298]: no change in test_data?

**Review by tabVersion — APPROVED** [review]
rest lgtm

**Comment by tabVersion** [comment]
@neverchanje please take a look at the syntax for the new types

File: e2e_test/source/kafka.slt (hunk 1 above)
Comment (waruto210) [reply e2e_test/source/kafka.slt:298]: `scripts/source/test_data/avro_bin.1` has been modified to add the `passed` data.

**Review by neverchanje — APPROVED** [review]
LGTM. I think this PR doesn't involve any additional syntaxes.

**Comment by hengm3467** [comment]
@waruto210 @tabVersion @neverchanje Quick question: Should we document the Avro schema that is supported in RW? Thanks.

**Pull request merged**